src/main/java/uk/gov/justice/digital/hmpps/keyworker/services/StaffSearch.kt

```diff
@@ -121,23 +129,60 @@ class StaffSearch(
     prisonCode: String,
     request: AllocatableSearchRequest,
   ): AllocatableSearchResponse {
-    val policy = AllocationContext.get().policy
+    val context = AllocationContext.get()
+    val reportingPeriod = ReportingPeriod.currentMonth()
+
+    val prisonConfig = prisonConfigRepository.findByCode(prisonCode) ?: PrisonConfiguration.default(prisonCode)
 
     val staffMembers =
       findAllocatableStaff(prisonCode)
         .filter { it.staffMember.matches(request.query) }
         .associateBy { it.staffMember.staffId }
 
-    val staffIdStrings = staffMembers.keys.map { it.toString() }.toSet()
+    val applicableAllocationsByStaff =
+      staffMembers.keys.associateWith {
+        allocationRepository
```

Comment:
This will be called once for every key worker in the prison when the page first loads. How about retrieving all the active allocations for a list of staff ids in the date range and filtering+associating them in memory?